src/inmanta/deploy/state.py

```diff
@@ -36,15 +38,45 @@ def requires_view(self) -> Mapping[ResourceIdStr, Set[ResourceIdStr]]:
     def provides_view(self) -> Mapping[ResourceIdStr, Set[ResourceIdStr]]:
         return self.reverse_mapping()
 
+    def get_all_provides_transitively(self, resource: ResourceIdStr | set[ResourceIdStr]) -> list[ResourceIdStr]:
+        """
+        This method returns all the provides (transitively) of the given resource.
+
+        :param resource: The resource or set of resources for which the provides have to be resolved transitively.
+        """
+        input_set = {resource} if isinstance(resource, str) else set(resource)
+        work: SimpleQueue[ResourceIdStr] = SimpleQueue()
```

Comment:
Do you know anything of the performance of this vs a simple list? I'm assuming this is better because it has efficient pops and appends. But it might also have more bookkeeping logic.

I ask because everything executed under the lock is somewhat performance-critical.

Reply:
I tested the performance and it seems that pop and push execute in O(1).

Reply:
I'm going to remember this